build_runner/lib/src/generate/build_impl.dart

```diff
@@ -568,7 +569,11 @@ class _SingleBuild {
       if (!_assetGraph.contains(assetId)) {
         throw new AssetNotFoundException(assetId);
       }
-      _assetGraph.get(assetId).isDeleted = true;
+      if (assetId != input) {
```

Comment:
I had written the interface for `deleteAsset` in what I had hoped was a future proof way. `PostProcessBuildStep` is safe - it only lets you delete the primary input. The interface we pass through is more general since I thought we could avoid future breaking changes if we did want to allow deleting other assets.

If we want we can clean this up with a point release of `package:build` and change `deleteAsset` to `deletePrimaryInput` with no arguments. Otherwise we can leave it as is and if we want to allow deleting other assets through the build step we can change the way we track deletes here.

Reply:
I think this is fine, we may very well want to allow deleting other assets in the future. I have also come up with at least one  reasonable use cases for reading other assets as well.